hazelcast/proxy/vector_collection.py

```diff
@@ -29,29 +29,117 @@
 
 
 class VectorCollection(Proxy["BlockingVectorCollection"]):
+    """VectorCollection contains documents with vectors.
+
+    Concurrent, distributed, observable and searchable vector collection.
+    The vector collection can work both async(non-blocking) or sync(blocking).
+    Blocking calls return the value of the call and block the execution until return value is calculated.
+    However, async calls return ``Future`` and do not block execution.
+    Result of the ``Future`` can be used whenever ready.
+    A ``Future``'s result can be obtained with blocking the execution by calling ``future.result()``.
```

Comment:
It can also have a callback function to handle it full async, so I believe we should give a link to the Future related documentation from here and that's enough.

"However, async calls return ``Future`` and do not block execution." and a link should be enough

Reply:
That's straight from the `IMap` docs.